step-capstone/src/App.js

```diff
@@ -1,27 +1,55 @@
 import React from 'react';
 import './styles/App.css';
-import { FirebaseContext } from './components/Firebase';
-import TravelObject from './components/TravelObject'
+import {FirebaseContext} from './components/Firebase';
+import {Button} from 'react-bootstrap';
 
+class App extends React.Component {
 
+  static contextType = FirebaseContext;
+  constructor() {
+    super();
+    this.state = {authState: false, userInfo: null};
+  }
 
-class App extends React.Component {
-  SomeComponent = () => (
-    <FirebaseContext.Consumer>
-      {(firebase) => {
-        firebase.creatFirebaseWidget();
-      }}
-    </FirebaseContext.Consumer>
-  );
-  value = this.SomeComponent();
+  afterAuthStateCheck(status) {
+    console.log(status);
+    if (status.signInStatus) {
+      this.setState({authState: true, userInfo: status.userInfo});
+      console.log("You have logged in.");
+    }
+    else {
+      this.setState({authState: false, userInfo: null});
+      console.log("You havent logged in yet");
+      this.context.createFirebaseWidget();
+    }
+  }
+
+  componentDidMount() {
+    console.log(this.context);
+    this.context.getUserInfo().then( (status) => this.afterAuthStateCheck(status));
+
+  }
   render () {
-    return(
-      <div className="App">
-        <p> Hello World </p>
-        {this.SomeComponent()}
+    console.log(this.state);
+    const display = (this.state.authState)? "block" : "none";
+    console.log(display);
+    const displayObject = {display: display};
+    return (
+      <div>
+          <div id="firebaseui-auth-container"></div>
```

Comment:
Since the App component acts as a router to different components (ex. login, home, etc) the login/logout UI should probably be their own components that the app can route to for specific URLS (and we'd set the URL depending on login state as you've written above).